src/App/App.js

```diff
@@ -35,10 +38,24 @@ class App extends Component {
     }
   }
 
-  showMovieDetails(id) {
-    this.setState({
+  findUserRatingsForFoundMovie() {
+    const userRating = this.state.userRatings.ratings.find(rating => {
+      return this.state.foundMovieId.id === rating.movie_id
+    })
+    if (userRating === undefined && this.state.isLoggedIn) {
+      this.setState({foundMovieRating: 'You haven\t rated this movie yet'})
+    } else if (!this.state.isLoggedIn) {
+      this.setState({foundMovieRating: 'Log in to rate this movie'})
+    } else {
+      this.setState({foundMovieRating: `You rated this movie: ${userRating.rating}`})
+    }
+  }
+
+  showMovieDetails = async (id) => {
+    await this.setState({
       foundMovieId: {id},
     })
+    this.findUserRatingsForFoundMovie()
     this.showMovieDetailsDisplay();
```

Comment:
I would generally not recommend waiting for a set state. Instead, just pass the id into the functions that need it.